appcelerator/titanium_mobile
# fix(ios): hierarchy error when using SplitWindow and NavigationWindow

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28497

**Summary:**
- Fixes regression introduced into Titanium 10.0.0 when you use both a `SplitWindow` and a `NavigationWindow` at the same time.
- Caused by nested `NavigationWindow` wrongly using root application controller instead of the 1sts controller available in the view hierarchy.

**Test:**
1. Build and run the below on iPad.
2. Verify the app does not crash on app startup. (This is the fix.)
3. Verify you see a `ListView` on the left side with a title "Master View" above it.
4. Verify you see a "Hello World" label in the middle of the screen with title "Detail View" above it.

```javascript
const masterWindow = Ti.UI.createWindow({ title: "Master View" });
masterWindow.add(Ti.UI.createListView({
	sections: [
		Ti.UI.createListSection({
			items: [
				{ properties: { title: "Row 1" } },
				{ properties: { title: "Row 2" } },
				{ properties: { title: "Row 3" } },
			],
		}),
	],
}));

const detailWindow = Ti.UI.createWindow({ title: "Detail View" });
detailWindow.add(Ti.UI.createLabel({ text: "Hello World!" }));

const splitWindow = Ti.UI.iOS.createSplitWindow({
	detailView: Ti.UI.createNavigationWindow({ window: detailWindow }),
	masterView: Ti.UI.createNavigationWindow({ window: masterWindow }),
	backgroundColor: "white",
	showMasterInPortrait: true,
});
Ti.UI.createNavigationWindow({ window: splitWindow }).open();
```
